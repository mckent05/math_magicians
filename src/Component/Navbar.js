import React from 'react';
import { Link } from 'react-router-dom';

const Navbar = () => (
  <div className="main-nav d-flex">
    <h2 className="header">Math Magicians</h2>
    <ul className="nav-links d-flex j-center a-center">
      <li>
        <Link to="/" className="nav-link" data-testid="home">Home</Link>
      </li>
      <li>
        <Link to="/calculator" className="nav-link" data-testid="calculator">Calculator</Link>
      </li>
      <li>
        <Link to="/quotes" className="nav-link" data-testid="quote">Quote</Link>
      </li>
    </ul>
  </div>
);

export default Navbar;
