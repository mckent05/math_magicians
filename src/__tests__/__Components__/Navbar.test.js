import { MemoryRouter } from 'react-router-dom';
import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import pretty from 'pretty';

import Navbar from '../../Component/Navbar';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

it('testing for the UI of Navbar', () => {
  act(() => {
    render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
      container,
    );
  });

  expect(pretty(container.innerHTML)).toMatchSnapshot();
});
