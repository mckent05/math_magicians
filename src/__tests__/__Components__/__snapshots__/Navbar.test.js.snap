// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing for the UI of Navbar 1`] = `
"<div class=\\"main-nav d-flex\\">
  <h2 class=\\"header\\">Math Magicians</h2>
  <ul class=\\"nav-links d-flex j-center a-center\\">
    <li><a class=\\"nav-link\\" data-testid=\\"home\\" href=\\"/\\">Home</a></li>
    <li><a class=\\"nav-link\\" data-testid=\\"calculator\\" href=\\"/calculator\\">Calculator</a></li>
    <li><a class=\\"nav-link\\" data-testid=\\"quote\\" href=\\"/quotes\\">Quote</a></li>
  </ul>
</div>"
`;
